deepfence_worker/utils/watermill.go

```diff
@@ -0,0 +1,137 @@
+package utils
+
+import (
+	"fmt"
+	"runtime/debug"
+
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/cenkalti/backoff/v3"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+)
+
+// RecoveredPanicError holds the recovered panic's error along with the stacktrace.
+type RecoveredPanicError struct {
+	V          interface{}
+	Stacktrace string
+}
+
+func (p RecoveredPanicError) Error() string {
+	return fmt.Sprintf("panic occurred: %#v, stacktrace: \n%s", p.V, p.Stacktrace)
+}
+
+// Recoverer recovers from any panic in the handler and appends RecoveredPanicError with the stacktrace
+// to any error returned from the handler.
+func Recoverer(h message.HandlerFunc) message.HandlerFunc {
+	return func(event *message.Message) (events []*message.Message, err error) {
+		panicked := true
```

Comment:
I don't think we need this?

Reply:
It is needed in case if any of the handler cause panic, this prevents other handlers from crashing along with the panic, others keep running and panic is recovered

Reply:
That's weird, panic unwinds the stack, so a `panic()` should be bound to a specific `recover()`

Reply:
Looks like there was a misunderstanding - I was talking about the boolean, not the whole function :)